src/clj/rems/db/entitlements.clj

```diff
@@ -36,6 +38,22 @@
                               :resource-ext-id resource-or-nil
                               :is-active? (not expired?)})))
 
+(defn- entitlement-to-permissions-api [{:keys [resid catappid start end mail userid approvedby]}]
+  (let [start-datetime (DateTime. start)]
+    (jwt/sign {:type "ControlledAccessGrants"
+                     :value (str "" resid)
+                     :source "https://ga4gh.org/duri/no_org"
+                     :by (str "" approvedby)
+                     :asserted (.getMillis start-datetime)} "secret"))) ;;TODO use key/real secret here
+
+(defn get-entitlements-for-permissions-api [user-or-nil resource-or-nil expired?]
+  {:ga4gh_visa_v1 (reduce conj [] (mapv entitlement-to-permissions-api
```

Comment:
`(reduce conj []` seems a bit redundant. doesn't `mapv` give you a vector already?